incubator/hnc/internal/validators/object.go

```diff
@@ -239,17 +242,19 @@ func (o *Object) handleInherited(op k8sadm.Operation, newSource, oldSource strin
 		return deny(metav1.StatusReasonForbidden, "Cannot create objects with the label \""+api.LabelInheritedFrom+"\"")
 
 	case k8sadm.Delete:
-		if o.isDeletingNS(oldInst) {
-			// There are few things more irritating in (K8s) life than having some stupid controller stop
-			// your namespace from being deleted. If there's an object in here and we've decided that the
-			// namespace should be deleted, then don't block anything!
-			//
-			// It's probably slightly safer to create a K8s client, actually load the namespace and check
-			// its deletion timestamp here. But it's much easier to just record this information in the
-			// namespace/HC reconciler and read it here.
-			return allow("allowing deletion of propagated object since namespace is being deleted")
+		// There are few things more irritating in (K8s) life than having some stupid controller stop
+		// your namespace from being deleted. If there's an object in here and we've decided that the
+		// namespace should be deleted, then don't block anything!
+		isDeleting, err := o.isDeletingNS(ctx, oldInst.GetNamespace())
+		if err != nil {
+			return deny(metav1.StatusReasonForbidden, "Cannot delete object propagated from namespace \""+oldSource+"\" with error: "+err.Error())
```

Comment:
We're not really forbidding the change - it's more like we couldn't validate whether it was legal or not. Maybe `StatusReasonInternalError` would be a more appropriate error here (HTTP 500).